### DISCUSSION TITLE
Hard reload causing 404

### DISCUSSION THREAD

**Discussion by hairlesshobo** [shlinkio/shlink-web-client#1145] [General]
### shlink-web-client version

4.1.1

### How do you use shlink-web-client

Self-hosted

### Summary

OK so I recently deployed shlink to a linode running Debian 12. Very basic install, I am hosting the main app using RoadRunner with apache setup for reverse proxy and SSL termination. All is working great there. I have shlink-web-client setup as a subdomain in apache serving the static files protected with basic HTTP auth. All is working there too.. the only issue is that if I am on a server page (for example: `/server/53ef4862-27bc-4b8e-aeaf-b715ef0a4695/overview`) and I reload the page, I am presented with a 404. 

Now, I see that there is a `.htaccess` file in the root of the static file directory, I verified that the rewrite module is enabled in apache using `a2enmod`, I made sure to add `AllowOverride all` to the vhost config.. everything seems to be in place but apache doesn't seem to be honoring the fallback of serving up `index.html` if a nonexistent page is requested. I also verified that apache is at least reading the htaccess file because if I intentionally modify the rewrite conditions, I can cause the app to break altogether (again, just for confirmation that mod rewrite is actually loaded).

I feel like its something simple and I am just overlooking it but I have no idea what at this point. Can anyone provide me with any suggestions of additional things to check?

**Comment by acelaya** [comment]
This is the nginx configuration used by shlink-web-client's docker image, to ensure any path falls back to the index.html file if it doesn't match an actual static file: https://github.com/shlinkio/shlink-web-client/blob/develop/config/docker/nginx.conf#L34

It's not Apache, but maybe it gives you some clue.

**Comment by hairlesshobo** [comment]
Well that did give me the idea to just cheat and use nginx to host it. When hosting with nginx, the fallback to `index.html` works as expected. Must be some weirdness with Apache on my system or something but whatever, got it fixed easy enough by swapping to nginx. Thanks!

**Reply by acelaya** [reply]
To be honest, I may have neglected support for Apache a bit, as I have not used it personally for quite some time. I didn't even remember this project shipped with its own `.htaccess` 😅 

I'll try to do a bit of experimentation and see if I can extend that `.htaccess` file so that this falling back to index.html logic works out of the box, or if that's not easy, at least document how to do it in the vhost configuration.

**Reply by hairlesshobo** [reply]
no worries at all. it was super quick to spin up nginx and just use apache to proxy to it. 

i was too lazy to reconfigure ssl, proxy, and auth in nginx lol so instead just kept apache as main and proxy to nginx for the admin page. simple fix and working great. 

thanks for your hard work on this tool!
